### PR TITLE
Add TRT EP memory leak test into trt perf script

### PR DESCRIPTION
**Pull request by chilo-ms**
**Description**: Add TRT EP memory test into trt perf script.
We need to add support of memory test, especially memory leak detection, into trt perf script in order to find memory issues before hand. It will be breakdown to two phases:

1.  Use address sanitizer to detect memory leak and then integrate this feature into perf script as well as CI pipeline. (This PR)
2.  Post result to Power BI (Next PR)

### CODE REVIEW

**Comment by chilo-ms** [comment]
Here is the test of memory test on CI:
https://dev.azure.com/onnxruntime/onnxruntime/_build/results?buildId=428773&view=logs&j=ef76da22-b066-5549-bf2e-8b7ec49c0b60&t=cadf3774-0095-5383-513d-972dd08cd4d8

File: onnxruntime/python/tools/tensorrt/perf/mem_test/main.cpp [hunk 1]
```diff
@@ -0,0 +1,385 @@
...
+//
+
+#include <assert.h>
+#include <vector>
+#include <onnxruntime_cxx_api.h>
+#include <tensorrt_provider_factory.h>
+
+std::unique_ptr<OrtTensorRTProviderOptions> get_default_trt_provider_options() {
+  auto tensorrt_options = std::make_unique<OrtTensorRTProviderOptions>();
+  tensorrt_options.get()->device_id = 0;
```
Comment (jywu-msft) [inline onnxruntime/python/tools/tensorrt/perf/mem_test/main.cpp:12]: don't need get()
just do tensorrt_options->device_id  
same with rest of file.

**Comment by jywu-msft** [comment]
in the longrun, see if we can integrate the leakcheck into the TensorRT Linux CI   (enabled for debug builds)
it would be nice to catch new mem leaks before code gets checked in.

File: onnxruntime/python/tools/tensorrt/perf/mem_test/main.cpp (hunk 1 above)
Comment (chilo-ms) [reply onnxruntime/python/tools/tensorrt/perf/mem_test/main.cpp]: Rewrote the code.

**Comment by chilo-ms** [comment]
@jywu-msft regarding integrating mem leak check into TRT EP CI, I think it's feasible. Will do it later.

**Review by jywu-msft — APPROVED** [review]

**Pull request merged**